daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/CompiledPackages.scala

```diff
@@ -34,6 +34,13 @@ private[lf] final class PureCompiledPackages(
     compilerConfig: Compiler.Config,
 ) extends CompiledPackages(compilerConfig) {
   override def getDefinition(dref: SDefinitionRef): Option[SDefinition] = defns.get(dref)
+
+  def copy(
+      packageIds: Set[PackageId] = packageIds,
+      pkgInterface: PackageInterface = pkgInterface,
+      defns: Map[SDefinitionRef, SDefinition] = defns,
+      compilerConfig: Compiler.Config = compilerConfig,
+  ) = new PureCompiledPackages(packageIds, pkgInterface, defns, compilerConfig)
```

Comment:
Maybe just better to make it case class.

Reply:
Ah, I was under the impression I couldn't do this, because of the existing companion object, but it appears it overrides it?